numba_rvsdg/core/transformations.py

```diff
@@ -290,37 +291,99 @@ def find_tail_blocks(
     return tail_subregion
 
 
-def extract_region(scfg, region_blocks, region_kind):
+def extract_region(scfg: SCFG, region_blocks, region_kind, parent_region: RegionBlock):
     headers, entries = scfg.find_headers_and_entries(region_blocks)
     exiting_blocks, exit_blocks = scfg.find_exiting_and_exits(region_blocks)
     assert len(headers) == 1
     assert len(exiting_blocks) == 1
     region_header = next(iter(headers))
     region_exiting = next(iter(exiting_blocks))
 
+    # Generate a new region name
+    region_name = scfg.name_gen.new_region_name(region_kind)
     head_subgraph = SCFG(
         {name: scfg.graph[name] for name in region_blocks}, name_gen=scfg.name_gen
     )
 
-    if isinstance(scfg[region_exiting], RegionBlock):
-        region_exiting = scfg[region_exiting].exiting
-    else:
-        region_exiting = region_exiting
-
-    subregion = RegionBlock(
-        name=region_header,
+    def update_exiting(region_block: RegionBlock):
```

Comment:
This needs a unit test.

Reply:
I don't have a way to test this function directly right now, because of lack of ability to form Regions using yaml like we do in `test_transforms`

Reply:
OK, maybe you can run `pytest` with `coverage` such that we can check if the given test-suite at least exercises all of the code paths in here?